### PR TITLE
Use flagutil.KubernetesOptions to set kube flags

### PR DESCRIPTION
**Pull request by hongkailiu**
Will rebase after https://github.com/openshift/ci-tools/pull/2306 and https://github.com/openshift/ci-tools/pull/2310

### CODE REVIEW

File: cmd/cluster-display/main.go [hunk 1]
```diff
@@ -34,7 +34,8 @@ func gatherOptions() (options, error) {
 	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
 	fs.StringVar(&o.logLevel, "log-level", "info", "Level at which to log output.")
 	fs.IntVar(&o.port, "port", 8090, "Port to run the server on")
-	fs.StringVar(&o.hiveKubeconfigPath, "hive-kubeconfig", "", "Path to the kubeconfig file to use for requests to Hive.")
+	o.kubernetesOptions.NOInClusterConfigDefault = true
```
Comment (hongkailiu) [inline cmd/cluster-display/main.go:37]: Need to fix the deployment since we changed the arg name.

File: cmd/ci-secret-bootstrap/main.go [hunk 2]
```diff
@@ -81,9 +80,8 @@ func parseOptions(censor *secrets.DynamicCensor) (options, error) {
 	fs.BoolVar(&o.validateItemsUsage, "validate-bitwarden-items-usage", false, fmt.Sprintf("If set, the tool only validates if all fields that exist in Vault and were last modified before %d days ago are being used in the given config.", allowUnusedDays))
 	fs.BoolVar(&o.dryRun, "dry-run", true, "Whether to actually create the secrets with oc command")
 	fs.BoolVar(&o.confirm, "confirm", true, "Whether to mutate the actual secrets in the targeted clusters")
-
-	fs.StringVar(&o.kubeConfigPath, "kubeconfig", "", "Path to the kubeconfig file to use for CLI requests.")
-	fs.StringVar(&o.kubeConfigDir, "kubeconfig-dir", "", "Path to the directory containing kubeconfig files to use for CLI requests.")
+	o.kubernetesOptions.NOInClusterConfigDefault = true
```
Comment (stevekuznetsov) [inline cmd/ci-secret-bootstrap/main.go:83]:
```go
o := options{
    kubernetesOptions: flagutil.KubernetesOptions{NoInClusterConfigDefault: true},
}
```

File: cmd/cluster-display/main.go (hunk 1 above)
Comment (stevekuznetsov) [reply cmd/cluster-display/main.go:37]: No breaking changes. Support both flags, then migrate.

**Review by stevekuznetsov — COMMENTED** [review]
The blast radius of this needs to be smaller

File: cmd/ci-secret-bootstrap/main.go (hunk 2 above)
Comment (hongkailiu) [reply cmd/ci-secret-bootstrap/main.go]: Updated.

File: cmd/cluster-display/main.go (hunk 1 above)
Comment (hongkailiu) [reply cmd/cluster-display/main.go]: Updated.

**Comment by hongkailiu** [comment]
/hold
will watch the deployment.

File: cmd/ci-secret-bootstrap/main_test.go [hunk 3]
```diff
@@ -447,6 +414,11 @@ func TestCompleteOptions(t *testing.T) {
 		},
 	}
 	for _, tc := range testCases {
+		kubernetesOptions, err := getKubernetesOptions(kubeConfigPath)
```
Comment (stevekuznetsov) [inline cmd/ci-secret-bootstrap/main_test.go:417]: What am I even looking at? What is this testing? Why is this here?

**Review by stevekuznetsov — COMMENTED** [review]
"Blast radius needs to be smaller" === "do not make a potentially fatal change to every component we own in one PR"

Start with one, then merge them in one by one as they work.

File: cmd/ci-secret-bootstrap/main_test.go (hunk 3 above)
Comment (hongkailiu) [reply cmd/ci-secret-bootstrap/main_test.go:417]: I do not have a setter on `kubernetesOptions` for `kubeconfig`.
It has to be parsed from args.
